### PR TITLE
feat: [FC-0047] Account Settings and Profile split

### PR DESCRIPTION
**Pull request by IvanStepanok**
Split account settings and the user profile

Design: https://www.figma.com/file/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.1?type=design&node-id=0-1&mode=design

## Quick demo:
Setting button on Discover, Dashboard and Profile screens.
ezgif-1-d38260a5e3

## Light and Dark themes:
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 58 20 Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 58 17

## Settings screen:
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 58 59 Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 59 00

## Manage account:
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 12 03 53 Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 12 03 51

## Landscape:
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 59 30
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-19 at 11 59 22

## Ipad:
Simulator Screenshot - iPad mini (6th generation) - 2024-04-19 at 11 59 12 Simulator Screenshot - iPad mini (6th generation) - 2024-04-19 at 11 59 09

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @IvanStepanok! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: default_config/dev/config.yaml
Comment (volodymyr-chekyrta) [inline default_config/dev/config.yaml:5]: Seems that a undesired change, revert it.

File: Authorization/Authorization/Presentation/Login/SignInView.swift
Comment (volodymyr-chekyrta) [inline Authorization/Authorization/Presentation/Login/SignInView.swift:30]: How about renaming it to headerBackground?

File: Core/Core/SwiftGen/Assets.swift
Comment (volodymyr-chekyrta) [inline Core/Core/SwiftGen/Assets.swift:115]: Since we don't use the word `icon` for icons, how about just renaming it to `settings`?

**Comment by volodymyr-chekyrta** [comment]
@IvanStepanok, please check this, the profile data in ProfileView does not change if I change it on the manage account screen.

https://github.com/openedx/openedx-app-ios/assets/127732735/d33b2499-5664-4dc9-a07c-d9e78829efb2

File: Profile/Profile/Presentation/Profile/ProfileView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Profile/ProfileView.swift:150]: Move to resources

File: Profile/Profile/Presentation/Settings/SettingsView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Settings/SettingsView.swift:110]: Remove extra space

File: Profile/Profile/Presentation/Settings/SettingsView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Settings/SettingsView.swift:124]: Please remove extra space

File: Profile/Profile/Presentation/Settings/SettingsView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Settings/SettingsView.swift:206]: Please format each parameter on its own line.

File: Profile/Profile/Presentation/Settings/SettingsView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Settings/SettingsView.swift:190]: Please format each parameter on its own line.

File: Profile/Profile/Presentation/Settings/ManageAccountView.swift
Comment (volodymyr-chekyrta) [inline Profile/Profile/Presentation/Settings/ManageAccountView.swift:46]: Please format each parameter on its own line.

**Comment by forgotvas** [comment]
Please add navigation titles for each controller and fix long tap back navigation:


https://github.com/openedx/openedx-app-ios/assets/480059/7c057280-dd18-4c0d-9450-e7185ff40261

**Comment by forgotvas** [comment]
Is that design decision to hide info on profile overview? In develop build i can see "year of birth" and "bio", in that branch only bio:

develop | profile | Manage account
--|--|--
Simulator Screenshot - iPad (10th generation) - 2024-04-22 at 14 52 21|Simulator Screenshot - iPad (10th generation) - 2024-04-22 at 14 52 22|Simulator Screenshot - iPad (10th generation) - 2024-04-22 at 14 52 42

**Comment by IvanStepanok** [comment]
Hi @forgotvas and thanks for the review!
The year of birth was removed by design, it's true, I double-checked it:
https://www.figma.com/file/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.1?type=design&node-id=0-1&mode=design

I fixed the back buttons and titles in the video settings too. Thanks!🙏

https://github.com/openedx/openedx-app-ios/assets/128456094/1c58c108-68f7-4c21-9385-b3846fa159fc

**Review by forgotvas — APPROVED** [review]
LGTM 👍

**Comment by shafqat-muneer** [comment]
Started reviewing PR

**Comment by shafqat-muneer** [comment]
The location of the back button varies across screens in landscape mode.

| Screens |
|----------|
| Simulator Screenshot - iPhone 12 (17 2) - 2024-04-26 at 16 21 46 |
| Simulator Screenshot - iPhone 12 (17 2) - 2024-04-26 at 16 30 26 |
| Simulator Screenshot - iPhone 12 (17 2) - 2024-04-26 at 16 30 42 |

File: Profile/Profile/Presentation/Profile/ProfileView.swift
Comment (shafqat-muneer) [inline Profile/Profile/Presentation/Profile/ProfileView.swift:149]: It seems, we can update like:
```
        if let bio = viewModel.userModel?.shortBiography, bio != "" {
            VStack(alignment: .leading, spacing: 6) {
                Text(ProfileLocalization.about) 
                    .font(Theme.Fonts.titleSmall)
                    .foregroundColor(Theme.Colors.textPrimary)
                    .accessibilityIdentifier("profile_info_text")
                
                    Text(bio)
                        .font(Theme.Fonts.bodyMedium)
                        .foregroundColor(Theme.Colors.textPrimary)
                        .accessibilityIdentifier("bio_text")
            }
```

File: Profile/Profile/Presentation/Profile/ProfileView.swift
Comment (shafqat-muneer) [inline Profile/Profile/Presentation/Profile/ProfileView.swift:208]: Please discard this change.

File: Profile/Profile/Presentation/Profile/ProfileView.swift
Comment (shafqat-muneer) [inline Profile/Profile/Presentation/Profile/ProfileView.swift:217]: Please discard this change.

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]
The review is currently underway. Kindly attend to the suggested changes made up to this stage.

File: OpenEdX/Router.swift
Comment (IvanStepanok) [reply OpenEdX/Router.swift:698]: Thank you for your suggestion to use a guard-let to avoid potential crashes by force unwrapping. While this is generally good practice, in this specific context, the force unwrap is intentional. Our application's design guarantees that the SettingsViewModel will always be present in the container when this function is called. Introducing a guard-let would imply a possibility of SettingsViewModel not being available, which contradicts our current architecture and could potentially confuse future maintainers about the assumptions in our codebase. Let's keep the force unwrap as an assertion of our design guarantees.

**Comment by IvanStepanok** [comment]
Thank you, @shafqat-muneer, for the review! All requests have been satisfied. Ready for another pass 🙌

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]

**Comment by shafqat-muneer** [comment]
@IvanStepanok Thank you for considering the feedback. 🎉 
Overall, I have a few additional minor observations that might require attention.

- Observed overlapping on manage account and settings screens

| Settings | Manage Account |
|-|-|
| Simulator Screenshot - iPhone 15 - 2024-05-01 at 12 58 06 | Simulator Screenshot - iPhone 15 - 2024-05-01 at 12 58 10 |

- Safe area is visible in **Video download quality** screen and it's covered in **Video settings** screen.

| Video download quality | Video settings |
|-|-|
| Simulator Screenshot - iPhone 15 - 2024-05-01 at 12 51 01 | Simulator Screenshot - iPhone 15 - 2024-05-01 at 12 51 04 |

- The position of the back arrow appears to be incorrect in comparison to the other screens.
Simulator Screenshot - iPhone 15 - 2024-05-01 at 13 01 14


- Finally, although it may not be directly related to the changes made in this pull request, I noticed it while reviewing the PR. We might want to address it in a separate PR. In the pre-login user experience, the course details aren't opening; it consistently shows an error.

https://github.com/openedx/openedx-app-ios/assets/11990137/c138dbdd-5c6a-4f7a-8f8c-4909858a058f

**Comment by IvanStepanok** [comment]
Hi, @shafqat-muneer, and thanks for the review! All problems are fixed. 🙌
Simulator Screenshot - iPhone 15 - 2024-05-03 at 15 59 11
Simulator Screenshot - iPhone 15 - 2024-05-03 at 15 59 09
Simulator Screenshot - iPhone 15 - 2024-05-03 at 15 59 06
Simulator Screenshot - iPhone 15 - 2024-05-03 at 15 37 24
Simulator Screenshot - iPhone 15 - 2024-05-03 at 15 37 22
Simulator Screenshot - iPhone 15 - 2024-05-04 at 14 05 12

**Comment by shafqat-muneer** [comment]
@IvanStepanok Thanks for addressing the feedback. 🎉 
All changes LGTM 👍. PR is approved.

It seems minor indentation / formatting related comments are missed. Please also address them before merge.
- [Point#1](https://github.com/openedx/openedx-app-ios/pull/397#discussion_r1584913187)
- [Point#2](https://github.com/openedx/openedx-app-ios/pull/397#discussion_r1584946972)
- [Point#3](https://github.com/openedx/openedx-app-ios/pull/397#discussion_r1584977187)

Separate issue is create [here](https://github.com/openedx/openedx-app-android/issues/307) for this observed problem.
> Finally, although it may not be directly related to the changes made in this pull request, I noticed it while reviewing the PR. We might want to address it in a separate PR. In the pre-login user experience, the course details aren't opening; it consistently shows an error.

**Review by shafqat-muneer — DISMISSED** [review]

**Comment by IvanStepanok** [comment]
Hi, @shafqat-muneer, please take a look 🙌

**Comment by shafqat-muneer** [comment]
Hi @IvanStepanok
Thanks
LGTM 👍

**Review by shafqat-muneer — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@IvanStepanok 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-ios&entry.867055334=https://github.com/openedx/openedx-app-ios/pull/397&entry.1484655318=https://github.com/IvanStepanok&entry.752974735=2024-04-17+19:14&entry.1917517419=2024-05-07+13:14&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
